hazelcast/src/main/java/com/hazelcast/datastore/JdbcDataStoreFactory.java

```diff
@@ -33,23 +34,26 @@ public class JdbcDataStoreFactory implements ExternalDataStoreFactory<DataSource
     private static final String JDBC_USERNAME = "jdbc.username";
     private static final String JDBC_PASSWORD = "jdbc.password";
     private ExternalDataStoreConfig config;
+    private DataSource shareDataSource;
 
     public void init(ExternalDataStoreConfig config) {
         this.config = config;
+        if (config.isShared()) {
+            shareDataSource = createDataSource();
+        }
     }
 
     @Override
     public DataSource getDataStore() {
+        return config.isShared() ? shareDataSource : createDataSource();
+    }
+
+    private DataSource createDataSource() {
         String jdbcUrl = config.getProperty(JDBC_URL);
-        if (config.isShared()) {
-            throw new UnsupportedOperationException("shared flag is not yet supported");
-        }
-        if (config.getProperties().containsKey(JDBC_USERNAME) || config.getProperties().containsKey(JDBC_PASSWORD)) {
-            String username = config.getProperty(JDBC_USERNAME);
-            String password = config.getProperty(JDBC_PASSWORD);
-            return new DataSourceFromJdbcUrl(jdbcUrl, username, password);
-        }
-        //TODO Use HikariDataSource for pooling or similar
-        return new DataSourceFromJdbcUrl(jdbcUrl);
+        HikariConfig dataSourceConfig = new HikariConfig();
```

Comment:
Hikari's default pool size is 10, do we want to expose more internal settings of data sources to the users? (see https://github.com/brettwooldridge/HikariCP#gear-configuration-knobs-baby)?